### PR TITLE
Created Dockerfile for Docker Image

### PR DESCRIPTION
**Pull request by GopalVerma1303**
Hey!

- [x] Create Docker image using Dockerfile
- [ ] Deploy Docker image to VPS/VDS for automatic deployment

### Steps to create Docker image

1. [Install Docker](https://docs.docker.com/get-docker/) on your machine.
1. Build your container: `docker build -t would-you-bot-website .`.
1. Run your container: `docker run -p 2123:2123would-you-bot-website`.

You can view your images created with `docker images`.

Please review and provide any feedback. Thank you!

### CODE REVIEW

**Review by mezotv — CHANGES_REQUESTED** [review]
Could you remove the package-lock file and get rid of anything that isnt part of the focker integration?

**Comment by GopalVerma1303** [comment]
Sure

**Comment by mezotv** [comment]
I noticed you have the vps/vds workflow unchecked is this something you are wanting to add to this aswell?

**Comment by GopalVerma1303** [comment]
> I noticed you have the vps/vds workflow unchecked is this something you are wanting to add to this aswell?

Absolutely, I'd love to incorporate the VPS/VDS workflow. Could we create a new issue to handle this separately and keep our current focus intact? Thanks!

**Comment by mezotv** [comment]
> > I noticed you have the vps/vds workflow unchecked is this something you are wanting to add to this aswell?
> 
> Absolutely, I'd love to incorporate the VPS/VDS workflow. Could we create a new issue to handle this separately and keep our current focus intact? Thanks!
> 

Sure just create an issue and I will assign it to you!

**Comment by GopalVerma1303** [comment]
I accidentally closed the merge request. Could you please re-review and merge it into the main branch? Thanks!

**Comment by mezotv** [comment]
> I accidentally closed the merge request. Could you please re-review and merge it into the main branch? Thanks!

Let me test it together with the auto deploy workflow then I will!

**Comment by GopalVerma1303** [comment]
Could you please specify the hosting provider and share instructions on how to obtain the required credentials for the deployment? Thanks!

**Comment by mezotv** [comment]
For the workflow this might be a good Inspiration https://github.com/EddieHubCommunity/BioDrop/blob/main/.github/workflows/publish-ghcr.yml

**Comment by mezotv** [comment]
> Hey! I've created Issue #46 but didn't have permission to add labels. Could you please add the "hacktoberfest" label? Thanks!

Alright there you go!

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+USER nextjs
+
+EXPOSE 3000
+
+ENV PORT 3000
+# set hostname to localhost
+ENV HOSTNAME "0.0.0.0"
+
+CMD ["node", "server.js"]
```
Comment (mezotv) [inline Dockerfile:83]: What is that supposed to do? Server.js isn't a file we have.

**Review by mezotv — CHANGES_REQUESTED** [review]
In the code I saw port 3000 being used a lot is it possible to switch that since our production server uses port 2123 for the website.

File: Dockerfile (hunk 1 above)
Comment (GopalVerma1303) [reply Dockerfile:83]: The server.js file in `.next/standalone/server.js` in Next.js is a minimal server that can be used to start your Next.js application in production. It is generated when you build your application with the `output: standalone` option in your next.config.js file.

**Comment by GopalVerma1303** [comment]
You can use this command run your container: `docker run -p 2123:2123 would-you-bot-website`

**Comment by mezotv** [comment]
> You can use this command to run your container: `docker run -p 2123:2123 would-you-bot-website`

Could you revert the changes you have done to the sitemap?

File: Dockerfile (hunk 1 above)
Comment (mezotv) [reply Dockerfile:83]: Id much rather not have this there as well as the standalone extension. Could you get rid of those?

File: next.config.js [hunk 2]
```diff
@@ -28,6 +28,8 @@ const nextConfig = {
       },
     ];
   },
+
+  output: 'standalone',
```
Comment (mezotv) [inline next.config.js:32]: Remove this please!

File: public/sitemap-0.xml [hunk 3]
```diff
@@ -1,15 +1,15 @@
...
+<url><loc>https://wouldyoubot.gg/legal-de</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/privacy</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/reddit</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/support</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/terms</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/vote</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
+<url><loc>https://wouldyoubot.gg/blog/welcome</loc><lastmod>2023-10-03T10:19:59.923Z</lastmod><changefreq>daily</changefreq><priority>0.7</priority></url>
 </urlset>
```
Comment (mezotv) [inline public/sitemap-0.xml:15]: Revert the changes here!

File: README.md [hunk 4]
```diff
@@ -6,10 +6,12 @@ The official repository for the [Would You website](https://wouldyoubot.gg)
...
-- [Introduction](#Introduction)
-- [Development](#development)
-- [About Bot](#about-Bot)
-- [Contributors](#contributors)
+- [Would You Web](#would-you-web)
+  - [Table of Contents](#table-of-contents)
+  - [Introduction](#introduction)
+  - [Development](#development)
+  - [About Bot](#about-bot)
+  - [Contributors](#contributors)
```
Comment (mezotv) [inline README.md:15]: Revert those changes please!

**Review by mezotv — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (GopalVerma1303) [reply README.md:15]: Sure

File: next.config.js (hunk 2 above)
Comment (GopalVerma1303) [reply next.config.js]: Sure

File: Dockerfile (hunk 1 above)
Comment (GopalVerma1303) [reply Dockerfile]: I have made some updates. Kindly review : [e78340f](https://github.com/Would-You-Bot/website/pull/41/commits/e78340f08c9b5e39663de7230647a70eb9533ea2)

**Comment by mezotv** [comment]
Will be checking this later thank you!

**Comment by mezotv** [comment]
Alright, I checked it and it seems to work well, tho I have a question how would the github workflow for this work?

**Comment by mezotv** [comment]
- Get rid of the redundant if statements for different lockfile types (npm, yarn, pnpm) since only one lockfile should be pushed to GitHub.

- Ensure that the code installs the necessary package manager (pnpm) before installing dependencies or building the app to avoid dependency conflicts.

- Consider changing file permissions after the build process to enhance security.

- Explicitly expose the port used by the application to ensure proper networking.

**Comment by mezotv** [comment]
Hey @GopalVerma1303, will the workflow be added to this pr as well?

**Comment by GopalVerma1303** [comment]
> Hey @GopalVerma1303, will the workflow be added to this pr as well?

Hey! I am planning to creating a new pr for this(Issue: #46) so as to keep our current focus intact.

**Comment by mezotv** [comment]
> > Hey @GopalVerma1303, will the workflow be added to this pr as well?
> 
> Hey! I am planning to creating a new pr for this(Issue: #46) so as to keep our current focus intact.

Hmm sounds good give me around an hour and I will test this!

**Review by mezotv — APPROVED** [review]
I got a bit confused with the whole server setup but I pushed it to the server and it's working well. Thanks for this pull request!

**Review by Podskio — APPROVED** [review]

**Pull request merged**